classes/UpgradeTools/ModuleAdapter.php

```diff
@@ -112,17 +113,37 @@ public function listModulesToUpgrade(array $modulesFromAddons)
         }
 
         foreach (scandir($dir) as $module_name) {
-            if (is_file($dir . DIRECTORY_SEPARATOR . $module_name)) {
+            // We don't update autoupgrade module
+            if ($module_name !== 'autoupgrade') {
```

Comment:
Am I misunderstanding something?
```suggestion
            if ($module_name === 'autoupgrade') {
```

Reply:
Fixed; Thanks